subsys/shell/modules/CMakeLists.txt

```diff
@@ -16,3 +16,11 @@ zephyr_sources_ifdef(
   CONFIG_DEVMEM_SHELL
   devmem_service.c
   )
+
+if(CONFIG_ENV_SHELL)
+  zephyr_library()
+  if (CONFIG_POSIX_ENV)
+    zephyr_library_compile_options(-U_POSIX_C_SOURCE -D_POSIX_C_SOURCE=200112L)
```

Comment:
Same comment about having this in the source

Reply:
I'm working on a better solution.

Putting it in the source is fine, but it's several more lines of code than necessary.